backend/src/v5/services/filesManager.js

```diff
@@ -214,4 +216,25 @@ FilesManager.storeFileStream = async (teamspace, collection, id, dataStream, met
 	await insertRef(teamspace, collection, { ...meta, ...refInfo, _id: id, mimeType });
 };
 
+FilesManager.getMD5FileHash = async (teamspace, container, filename) => {
+	const refEntry = await getRefEntry(teamspace, `${container}.history.ref`, filename);
+
+	if (Object.keys(refEntry).includes('MD5Hash')) {
+		return {
+			hash: refEntry.MD5Hash,
+			size: refEntry.size,
+		};
+	}
+
+	const file = await FilesManager.getFile(teamspace, `${container}.history`, filename);
```

Comment:
getting filesManager to append history is a bit too coupled (whilst we're only using it in history atm, we can in theory get md5 for anything).

I would rename container to collection and get the processor to pass in `${container}.history` instead